grails-app/domain/org/pih/warehouse/invoice/Invoice.groovy

```diff
@@ -145,6 +146,14 @@ class Invoice implements Serializable {
         return InvoiceStatus.PENDING
     }
 
+    boolean getIsPrepaymentInvoice() {
+        return invoiceType?.code == InvoiceTypeCode.PREPAYMENT_INVOICE
+    }
+
+    Float getTotalPrepaymentValue() {
+        return !isPrepaymentInvoice && invoiceItems.any { it.prepaymentItem } ? invoiceItems?.collect { it?.prepaymentItem?.totalAmount }?.sum() : 0
```

Comment:
I feel like I asked this one already, but don't see it anywhere. Why are we doing it this way? Why not just calculate this 
```
return (isPrepaymentInvoice) ? invoiceItems.sum { it.totalAmount } : 0
```
That way we remove the need for the N+1 query invoked by getPrepaymentItem().